### ISSUE TITLE
openInputStream for snapshot

### ISSUE THREAD

**Issue by mgorniew** [Azure/azure-storage-java#253]
Hi,

When I'm trying to use `openInputStream()` on `CloudBlob` snapshot I'm getting exception:

```
java.lang.IllegalArgumentException: Cannot perform this operation on a blob representing a snapshot.
	at com.microsoft.azure.storage.blob.CloudBlob.assertNoWriteOperationForSnapshot(CloudBlob.java:434)
	at com.microsoft.azure.storage.blob.CloudBlob.openInputStream(CloudBlob.java:2378)
	at com.microsoft.azure.storage.blob.CloudBlob.openInputStream(CloudBlob.java:2347)
```
Is this expected? Why I can't open InputStream to snapshot?

**Comment by mirobers** [comment]
This does seem to be a bug. We'll fix this for the next release. Thanks for reporting!

**Comment by mirobers** [comment]
This is fixed in 7.0.
